xwiki-platform-core/xwiki-platform-uiextension/xwiki-platform-uiextension-api/src/main/java/org/xwiki/uiextension/internal/WikiUIExtensionParameters.java

```diff
@@ -123,12 +127,18 @@ public WikiUIExtensionParameters(String id, String rawParameters, ComponentManag
      */
     private Map<String, String> parseParameters(String rawParameters)
     {
-        Map<String, String> result = new HashMap<String, String>();
-        for (String line : rawParameters.split("[\\r\\n]+")) {
-            String[] pair = line.split("=", 2);
-            if (pair.length == 2 && !"".equals(pair[0]) && !"".equals(pair[1])) {
-                result.put(pair[0], pair[1]);
+        Map<String, String> result;
+        Properties properties = new Properties();
+        try {
+            properties.load(new StringReader(rawParameters));
+            result = new HashMap<>();
+            for (String key : properties.stringPropertyNames()) {
+                if (!Objects.equals(key, "")) {
```

Comment:
Why not using something such as `StringUtils.isBlank(key)` here? Looks safer.

Reply:
Btw Properties extends Hashtable, so it could only have one element with "" as key, no? Wouldn't that work to directly create `new HashMap<>(properties)`?

Reply:
Properties inherits `Map<Object, Object>` and `parseParameters` returns `Map<String,String>` so doing `new HashMap<String,String>(properties)` does not typecheck.

Reply:
Indeed.